### ISSUE TITLE
Segmentation node ID issues with loading .seq.mrb

### ISSUE THREAD

**Issue by RMR54** [SlicerRt/Sequences#66]
After loading in a sequence metafile, two segmentations are added, one with draw tube effect drawn (example, "Seg_1"), and the other being empty (example, "Empty_Seg". A new synchronized sequence node is added to the volume browser node, with the Seg_1 set as data node at an index value (ex. "128.778s") sandwiched between previous and next indices at which "Empty_Seg" is set as data node

Sequence Node:
![image](https://user-images.githubusercontent.com/8135181/52296867-6e399980-294d-11e9-8eba-00b6887f1087.png)

Adding to synchronized node:
![image](https://user-images.githubusercontent.com/8135181/52296837-582bd900-294d-11e9-9fee-866c3a4e7346.png)

The purpose is to display the segmentation only at one index:
![image](https://user-images.githubusercontent.com/8135181/52296925-88737780-294d-11e9-8a12-a111b553d6cb.png)

and hide at other indices:
![image](https://user-images.githubusercontent.com/8135181/52296948-97f2c080-294d-11e9-853d-546f89544764.png)

Then I try to save the sequence as ".seq.mrb". 

When loading this file back in as "Sequence", I can see the sequences are successfully loaded back in (the data nodes and indices are maintained. However, the segment editor module shows no segments for the segmentation that had also been loaded in with the ".seq.mrb" load. 
![image](https://user-images.githubusercontent.com/8135181/52297297-6b8b7400-294e-11e9-9d8a-33081c4dd67c.png)

When adding observer for "slicer.vtkMRMLScene.NodeAddedEvent", It looked like the segmentation nodes added have different IDs compared to the sequence node data node IDs (for the same segmentation node names). Would this be a reason for not segment editor not recognizing the segments for a given segmentation node?

If I try to load this ".seq.mrb" as MRB Slicer Data Bundle, I can see the segments for the segmentation, but lose the data node and index info for the sequence node.

![image](https://user-images.githubusercontent.com/8135181/52297371-95449b00-294e-11e9-8d55-7e45fad26286.png)

What would be the best way to load in the .seq.mrb files back in such that segmentation, segments, sequence node, its data nodes and indices are all maintained and accessible?

This was observed in stable release Slicer 4.10.1 in Windows

**Comment by lassoan** [comment]
The seq.mrb file does not contain all necessary information to restore a sequence (indexName, indexUnit, indexType, indexValues/node ID pairs are all stored in the scene file). So, if you want to replay the segmentation sequence then you must load the original scene file and not just import the .seq.mrb file. In the future, we may change sequence storage so that all information is stored in the .seq.mrb file.

To record multiple segmentation states, make sure that in in Sequence browser module "Save changes" is enabled for the segmentation node.

If you just want to show/hide segments or change other display properties then you may record segmentation display node instead of segmentation node.

**Comment by RMR54** [comment]
Thank you for the clarification @lassoan! It looks like after loading in ".seq.mrb" and master volume node, if we set browser node to the specific index value that's defined in the sequence, then the segment editor gets updated and the segmentation node is also updated to show a segment below "Seg_1". 

Additionally, when the segmentation nodes are added to scene, they contain a different ID than the one specified by the sequence node, so I had to update the data nodes in sequence node. But this is specific to loading in sequence.seq.mrb, and is probably not required when loading in both original scene and .seq.mrb files.
